### ISSUE TITLE
Watch Request Blocked When Member Cluster Offline

### ISSUE THREAD

**Issue by xigang** [karmada-io/karmada#5672]
**What happened**:

When the member cluster goes offline, there is a scenario where the client's Watch request gets blocked and does not receive pod events.

**What you expected to happen**:

Should we set a timeout: Set a reasonable timeout for cache.Watch() calls using context.WithTimeout or context.WithDeadline to control the operation time?

https://github.com/karmada-io/karmada/blob/master/pkg/search/proxy/store/multi_cluster_cache.go#L354

example code:
```
        // 使用带超时的上下文
        watchCtx, cancel := context.WithTimeout(ctx, watchTimeout)
        defer cancel()

        w, err := cache.Watch(watchCtx, options)
        if err != nil {
            if err == context.DeadlineExceeded {
                klog.Warningf("Watch timeout for cluster %s: %v", cluster, err)
                continue // 跳过这个集群，继续处理下一个
            }
            mux.Stop()
            return nil, fmt.Errorf("failed to watch cluster %s: %v", cluster, err)
        }
```

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- Karmada version:
- kubectl-karmada or karmadactl version (the result of `kubectl-karmada version` or `karmadactl version`):
- Others:

**Comment by XiShanYongYe-Chang** [comment]
> When the member cluster goes offline, there is a scenario where the client's Watch request gets blocked and does not receive pod events.

Are events in other normal clusters affected?

**Comment by xigang** [comment]
@XiShanYongYe-Chang Unable to receive events from all member clusters through the aggregated apiserver, suspecting that the watch is blocked in `cache.watch()`.

**Comment by XiShanYongYe-Chang** [comment]
Thank you for your reply.

According to your method, the watch connection will be disconnected after a certain period of time, and then the client needs to initiate a watch request again. Do I understand it correctly?

**Comment by xigang** [comment]
> Thank you for your reply.
> 
> According to your method, the watch connection will be disconnected after a certain period of time, and then the client needs to initiate a watch request again. Do I understand it correctly?

Yes, but my scenario is quite special. The member cluster has gone offline, but since it hasn't been removed from the  `ResourceRegistry`, the rewatch requests still target the offline cluster, causing the watch requests to get stuck.

**Comment by XiShanYongYe-Chang** [comment]
Hi @xigang, why close this issue?

**Comment by xigang** [comment]
> Hi @xigang, why close this issue?

@XiShanYongYe-Chang I will submit a fix PR later.

**Comment by xigang** [comment]
> Hi @xigang, why close this issue?

@XiShanYongYe-Chang PR submitted. PTAL.

**Comment by RainbowMango** [comment]
> Yes, but my scenario is quite special. The member cluster has gone offline, but since it hasn't been removed from the ResourceRegistry, the rewatch requests still request the offline cluster, causing the watch requests to get stuck. I suspect this is the reason.

> I will conduct a test to verify.

Has this been confirmed? If so, they can use this case to reproduce it.
